src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -674,6 +674,15 @@ ModifyPartialQuerySupported(Query *queryTree, bool multiShardQuery,
 			{
 				Assert(hasVarArgument || hasBadCoalesce);
 			}
+
+			if (IsA((Node *) targetEntry->expr, FieldStore))
+			{
+				/* DELETE cannot do field indirection already */
+				Assert(commandType == CMD_UPDATE || commandType == CMD_INSERT);
+				return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
+									 "inserting or modifying composite type fields is not "
```

Comment:
can we give a hint on how to do  it?

Reply:
I will give a hint to prefer insert/update the whole composite type

Reply:
Does it look good ?
```c
				return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
									 "inserting or modifying composite type fields is not "
									 "supported", NULL,
									 "To workaround this, use whole column for inserting or "
									 "modifying the composite type fields");
```

Reply:
Maybe "Use the column name to insert or update the composite type as a single value"